### PR TITLE
Typed Builder implementation Part 8.1: Uninterpreted constants

### LINKED ISSUE #1918: Add uninterpreted constants to typed builder
The current version only support strings constructed via `str("...")` of type `StrT1`. We need a way to construct values of type `ConstT1`.

### PR DESCRIPTION
**Pull request by Kukovec**
- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [ ] Entries added to [./unreleased/][unreleased] for any new functionality

closes #1918

[unreleased]: https://github.com/informalsystems/apalache/tree/unstable/.unreleased

### CODE REVIEW

File: tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/LiteralAndNameBuilder.scala [hunk 1]
```diff
@@ -23,6 +23,12 @@ trait LiteralAndNameBuilder extends UnsafeLiteralAndNameBuilder {
   /** b: Bool */
   def bool(b: Boolean): TBuilderInstruction = _bool(b).point[TBuilderInternalState]
 
+  /** root_OF_A : A */
+  def const(root: String, A: ConstT1): TBuilderInstruction = _const(root, A).point[TBuilderInternalState]
+
+  /** v : A */
```
Comment (konnov) [inline tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/LiteralAndNameBuilder.scala:29]:
```suggestion
  /** v : Str */
```

**Review by konnov — APPROVED** [review]
Looks good. Just one comment.

File: tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/LiteralAndNameBuilder.scala (hunk 1 above)
Comment (Kukovec) [reply tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/LiteralAndNameBuilder.scala:29]: No, `v` has to have an uninterpreted type. For example `1_OF_A` (it's not `: Str`)

Comment (konnov) [reply tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/LiteralAndNameBuilder.scala:29]: ah, this was not clear to me. Shall we call the method `constParsed` or something?

**Pull request merged**